l10n_it_fatturapa_in/models/partner.py

```diff
@@ -47,3 +54,24 @@ class Partner(models.Model):
              "-1 to use the default precision",
         default=-1
     )
+
+    @api.constrains('vat', 'country_id')
+    def check_vat(self):
+        # Replicate e-invoice specifications check on IdFiscaleIVA:
+        # only italian VAT is checked
+        italy = self.env.ref('base.it')
+        it_partners = self.filtered(lambda p: p.country_id == italy)
+        super(Partner, it_partners).check_vat()
+        try:
+            super(Partner, self - it_partners).check_vat()
+        except ValidationError as ve:
+            if 'fatturapa_in_skip_no_it_vat_check' in self.env.context:
+                log_message = \
+                    "Validation Error skipped during e-bill import:\n" \
+                    "{exc_message}" \
+                    .format(
+                        exc_message=ve.args[0],
+                    )
+                _logger.info(log_message)
+            else:
+                raise ve
```

Comment:
Se il controllo deve essere fatto solo per partner ITA, perché non chiamare più semplicemente la super solo se il codice stato contiene IT? In tal modo si evita il check_vat per tutti gli altri e si evita di anche di gestire ulteriori parametri di contesto che di fatto non hanno un senso dal momento che il controllo dovrà sempre essere fatto solo per gli italiani, quindi non esistono casi reali in cui dovrà essere fatto a scelta dell'utente o di una specifica situazione.

Insomma:
```
    @api.constrains('vat', 'country_id')
    def check_vat(self):
        # Replicate e-invoice specifications check on IdFiscaleIVA:
        # only italian VAT is checked
        it_partners = self.filtered(lambda p: p.country_id.code == 'IT')
        super(Partner, it_partners).check_vat()
```

Reply:
Grazie della review!

> Se il controllo deve essere fatto solo per partner ITA, perché non chiamare più semplicemente la super solo se il codice stato contiene IT?

Senza parametri nel contesto, non potrei distinguere quando questo controllo viene chiamato durante la normale creazione di un partner da interfaccia, rispetto a quando viene chiamato durante l'importazione di una fattura elettronica.

> In tal modo si evita il check_vat per tutti gli altri e si evita di anche di gestire ulteriori parametri di contesto che di fatto non hanno un senso dal momento che il controllo dovrà sempre essere fatto solo per gli italiani, quindi non esistono casi reali in cui dovrà essere fatto a scelta dell'utente o di una specifica situazione.

Ci avevo pensato, ma preferisco chiamare il `super` di `check_vat` anche per i partner stranieri per non rompere la catena di ereditarietà del metodo.

> Insomma:
> 
> ```
>     @api.constrains('vat', 'country_id')
>     def check_vat(self):
>         # Replicate e-invoice specifications check on IdFiscaleIVA:
>         # only italian VAT is checked
>         it_partners = self.filtered(lambda p: p.country_id.code == 'IT')
>         super(Partner, it_partners).check_vat()
> ```

In questo modo, non verrebbe mai controllata la Partita IVA di un partner straniero, nemmeno durante la creazione del partner da interfaccia.

---
Ho comunque modificato il codice in modo da chiarire che succedono cose particolari solo quando c'è il parametro `fatturapa_in_skip_no_it_vat_check` nel contesto.